pkg/util/github/repo.go

```diff
@@ -140,5 +141,44 @@ func (c *Client) PushInitRepo(transitBranch, branch, localPath string) error {
 
 	// 2. push local path to repo
 	needRollBack, err := c.PushLocalPathToBranch(transitBranch, branch, localPath)
+	if err != nil {
+		return err
+	}
+
+	// 3. protect branch
+	err = c.ProtectBranch(branch)
 	return err
 }
+
+// ProtectBranch will protect the special branch
+func (c *Client) ProtectBranch(branch string) error {
+	req := &github.ProtectionRequest{
+		RequiredStatusChecks: nil,
+		EnforceAdmins:        false,
+		RequiredPullRequestReviews: &github.PullRequestReviewsEnforcementRequest{
+			RequireCodeOwnerReviews: true,
+			DismissStaleReviews:     true,
+		},
+		Restrictions: nil,
```

Comment:
I looked into github branch protection.

I think here is a configuration we can discuss that
1. `DismissStaleReviews` set to `false`, otherwise the reviewer will frequently approve, resulting in too much trouble to reviewer.
2. set `RequiredApprovingReviewCount` to `1`, so that PRs after approving can be merged, and the user can adjust them later.
3. set `RequiredConversationResolution` to `true`, all comments raised should be resolved.

Full config:
```go
req := &github.ProtectionRequest{
	RequiredPullRequestReviews: &github.PullRequestReviewsEnforcementRequest{
		RequireCodeOwnerReviews:      true,
		RequiredApprovingReviewCount: 1,
	},
	RequiredConversationResolution: github.Bool(true),
}
```

What's your opnion? @csonezp

Reply:
@iyear  I agree with you except for ```DismissStaleReviews```.
In my option, once a pull request is approved, if there are further commits, the pull request should go to un-approved state automatically.
This avoids unreviewed code being merged into the main branch.
Do I understand right?

Reply:
I agree with you. I've given it some thought, `DismissStaleReviews` is still set to `true`, we just go by the strictest rules and the user adjusts them.